Add empty JSON data file as empty map

acclassic opened the pull request:
This fixes issue #8601.

Test Case:
1. Create a new Hugo Project or use an existing one
2. Add an empty JSON file to the data folder

Result:
The site will not crash even when the data is accessed in the layout files like this: 
```
{{ range .Site.Data.jazz.bass }}
   {{ . }}
{{ end }}
```

This is my first pull request. So any suggestion for improvement is highly welcome.

bep commented:
Thanks. The fix looks good, but I think we need a test that demonstrates this issue to avoid someone breaking it in the future.

acclassic commented:
You're welcome. 
Thanks for the feedback. I added an empty JSON to the decoder_test file. If the Unmarshal function is now modified and does not check for the length of the data the test will fail.

acclassic commented:
Sorry I forgot to rebase my commit.

bep commented:
Thanks for this, closing this in favour of #10537

For future contributions I would recommend that you start by creating a new branch for the PR and not adding commits to your master branch. This makes for a much simpler work flow.
